pkg/sanity/resources.go

```diff
@@ -26,19 +27,62 @@ import (
 
 	"github.com/container-storage-interface/spec/lib/go/csi"
 
+	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
-// VolumeInfo keeps track of the information needed to delete a volume.
-type VolumeInfo struct {
+// managedResourceInfos is a map of managed resource infos keyed by resource
+// IDs.
+type resourceInfos map[string]resourceInfo
+
+// getLastInserted returns the last inserted resourceInfo, or nil if
+// resourceInfos is empty.
+func (ris resourceInfos) getLastInserted() *resourceInfo {
+	if len(ris) == 0 {
+		return nil
+	}
+
+	var last *resourceInfo
+	for _, ri := range ris {
+		if last == nil || ri.order > last.order {
+			last = &ri
+		}
+	}
+	return last
+}
+
+// addItem adds a resourceInfo item by the given ID and with the given data.
+func (ris resourceInfos) addItem(id string, data interface{}) {
+	nextOrder := -1
+	if lastInserted := ris.getLastInserted(); lastInserted != nil {
+		nextOrder = lastInserted.order
+	}
+	nextOrder++
+
+	ris[id] = resourceInfo{
+		id:    id,
+		order: nextOrder,
+		data:  data,
+	}
+}
+
+// resourceInfo represents a resource (i.e., a volume or a snapshot).
+type resourceInfo struct {
+	id    string
+	order int
+	data  interface{}
+}
+
+// volumeInfo keeps track of the information needed to delete a volume.
+type volumeInfo struct {
 	// Node on which the volume was published, empty if none
 	// or publishing is not supported.
 	NodeID string
-
-	// Volume ID assigned by CreateVolume.
-	VolumeID string
```

Comment:
I removed `VolumeID` because it's now part of the key in `resourceInfos`. Same considerations as with the unexporting of `[Vv]olumeInfo` apply, though.